src/main/java/com/hotels/bdp/circustrain/bigquery/table/service/partitioned/PartitionedTableService.java

```diff
@@ -68,23 +68,39 @@ public Table getTable() {
 
   @Override
   public List<Partition> getPartitions() {
-    return factory.generate(getPartitionKey(filteredTable), result.iterateAll());
+    return factory.generate(getPartitionKey(), getPartitionKeyType(), result.iterateAll());
   }
 
-  private String getPartitionKey(com.google.cloud.bigquery.Table table) {
-    Schema schema = table.getDefinition().getSchema();
+  private String getPartitionKey() {
+    Schema schema = filteredTable.getDefinition().getSchema();
     if (schema == null) {
       return "";
     }
 
     return sanitisePartitionKey(schema);
   }
 
+  private String getPartitionKeyType() {
+    if (tableContainsField()) {
```

Comment:
Yes, I was assuming that if there was at least one field in the schema, then it would be the `partitionedBy` field. But I will change it to make it more explicit that it's checking that `partitionedBy` exists in the schema.